otx/algorithms/detection/tasks/inference.py

```diff
@@ -226,18 +226,22 @@ def _init_recipe(self):
         if train_type not in (TrainType.SEMISUPERVISED, TrainType.INCREMENTAL):
             raise NotImplementedError(f"Train type {train_type} is not implemented yet.")
         if train_type == TrainType.SEMISUPERVISED:
-            if self._data_cfg.get("data", None) and self._data_cfg.data.get("unlabeled", None):
-                recipe = os.path.join(recipe_root, "semisl.py")
+            if self._is_training:
+                if self._data_cfg.get("data", None) and self._data_cfg.data.get("unlabeled", None):
+                    recipe = os.path.join(recipe_root, "semisl.py")
+                else:
+                    logger.warning("Cannot find unlabeled data.. convert to INCREMENTAL.")
+                    train_type = TrainType.INCREMENTAL
             else:
-                logger.warning("Cannot find unlabeled data.. convert to INCREMENTAL.")
-                train_type = TrainType.INCREMENTAL
+                recipe = os.path.join(recipe_root, "semisl.py")
```

Comment:
This also can cause issue if the train is not actually run in supervised mode. For example, even if we set train type as SEMISUPERVISED but can be run through incremental mode by if-statement above(unlabeled data actually does not exist). In that case, the right way is that running the inference is in incremental mode as well. But I guess it would run anyway even though we load whatever because it's just a inference. We can leave it as is but hope to resolve wisely. If you got some solution, please share. This is not serious but it would be nice if we can match the train type load between train and inference.

Reply:
This may be resolved by songki's suggestion above.